jujugui/static/gui/src/app/components/inspector-relations-item/inspector-relations-item.js

```diff
@@ -38,19 +52,77 @@ YUI.add('inspector-relations-item', function() {
       });
     },
 
+    /**
+      Handles the checkbox change action by either calling the parent supplied
+      whenChanged method and by setting the local checked state.
+
+      @method _handleChange
+      @param {Object} The change event from the checkbox.
+    */
+    _handleChange: function(e) {
+      var whenChanged = this.props.whenChanged;
+      var checked = e.currentTarget.checked;
+      this.setState({checked: checked}, () => {
+        // When whenChanged is set by the list parent and is used to (de)select
+        // all checkboxes. It is called in the setState callback so that the
+        // updated state is available if we inspect it from whenChanged.
+        if (whenChanged) {
```

Comment:
Might be worth adding some PropTypes to this component and setting whenChanged to isRequired.